crates/oxc_query/src/edges.rs

```diff
@@ -0,0 +1,2169 @@
+        resolve_info: &ResolveEdgeInfo,
+    ) -> ContextOutcomeIterator<'a, Vertex<'b>, VertexIterator<'a, Vertex<'b>>> {
+        super::import::entire_span(contexts, resolve_info)
+    }
+
+    pub(super) fn parent<'a, 'b: 'a>(
+        contexts: ContextIterator<'a, Vertex<'b>>,
+        _resolve_info: &ResolveEdgeInfo,
+        adapter: &'a Adapter<'b>,
+    ) -> ContextOutcomeIterator<'a, Vertex<'b>, VertexIterator<'a, Vertex<'b>>> {
+        super::parents(contexts, adapter)
+    }
+
+    pub(super) fn span<'a, 'b: 'a>(
+        contexts: ContextIterator<'a, Vertex<'b>>,
+        _resolve_info: &ResolveEdgeInfo,
+    ) -> ContextOutcomeIterator<'a, Vertex<'b>, VertexIterator<'a, Vertex<'b>>> {
+        super::get_span(contexts)
+    }
+
+    pub(super) fn specific_import<'a, 'b: 'a>(
+        contexts: ContextIterator<'a, Vertex<'b>>,
+        resolve_info: &ResolveEdgeInfo,
+    ) -> ContextOutcomeIterator<'a, Vertex<'b>, VertexIterator<'a, Vertex<'b>>> {
+        super::import::specific_import(contexts, resolve_info)
+    }
+}
+
+pub(super) fn resolve_interface_edge<'a, 'b: 'a>(
+    contexts: ContextIterator<'a, Vertex<'b>>,
+    edge_name: &str,
+    _parameters: &EdgeParameters,
+    resolve_info: &ResolveEdgeInfo,
+) -> ContextOutcomeIterator<'a, Vertex<'b>, VertexIterator<'a, Vertex<'b>>> {
+    match edge_name {
+        "entire_span" => interface::entire_span(contexts, resolve_info),
+        "extend" => interface::extend(contexts, resolve_info),
+        "name_span" => interface::name_span(contexts, resolve_info),
+        _ => {
+            unreachable!("attempted to resolve unexpected edge '{edge_name}' on type 'Interface'")
+        }
+    }
+}
+
+mod interface {
+    use std::rc::Rc;
+
+    use oxc_ast::ast::Expression;
+    use trustfall::provider::{
+        resolve_neighbors_with, ContextIterator, ContextOutcomeIterator, ResolveEdgeInfo,
+        VertexIterator,
+    };
+
+    use super::super::vertex::Vertex;
+    use crate::vertex::InterfaceExtendVertex;
+
+    pub(super) fn entire_span<'a, 'b: 'a>(
+        contexts: ContextIterator<'a, Vertex<'b>>,
+        _resolve_info: &ResolveEdgeInfo,
+    ) -> ContextOutcomeIterator<'a, Vertex<'b>, VertexIterator<'a, Vertex<'b>>> {
+        resolve_neighbors_with(contexts, |v| {
+            Box::new(std::iter::once(Vertex::Span(
+                v.as_interface()
+                    .unwrap_or_else(|| {
+                        panic!("expected to have an interface vertex, instead have: {v:#?}")
+                    })
+                    .interface
+                    .span,
+            )))
+        })
+    }
+
+    pub(super) fn extend<'a, 'b: 'a>(
+        contexts: ContextIterator<'a, Vertex<'b>>,
+        _resolve_info: &ResolveEdgeInfo,
+    ) -> ContextOutcomeIterator<'a, Vertex<'b>, VertexIterator<'a, Vertex<'b>>> {
+        resolve_neighbors_with(contexts, |v| {
+            #[allow(clippy::option_if_let_else)]
+            if let Some(extends) = &v
```

Comment:
this if statement may be too long, maybe it's worth extracting it out to a variable + we can use .into_iter() here I think

Reply:
I agree, use `.into_iter()` on the option.

Reply:
Converted the rest of if let Some's to into_iter here: [77f272f](https://github.com/web-infra-dev/oxc/pull/599/commits/77f272f08cbea3386bc520b3106cb3d569e4dca8)